File: endpoints/src/main/java/com/cromoteca/samples/spring/SpringEndpoint.java
package com.cromoteca.samples.spring;

import com.vaadin.flow.server.auth.AnonymousAllowed;
import com.vaadin.hilla.Endpoint;
import com.vaadin.hilla.Nonnull;
import org.springframework.beans.factory.annotation.Autowired;

@Endpoint
@AnonymousAllowed
public class SpringEndpoint {
  private final SpringDependency dependency;

  @Autowired
  public SpringEndpoint(SpringDependency dependency) {
    this.dependency = dependency;
  }

  @Nonnull
  public String hello() {
    return dependency.hello();
  }
}

File: generator/src/test/resources/com/cromoteca/samples/spring/SpringEndpoint.ts
import client from '../../../../connect-client.default.js';
import { type EndpointRequestInit } from '@vaadin/hilla-frontend';

async function hello(init?: EndpointRequestInit): Promise<string> {
    return client.call('com.cromoteca.samples.spring.SpringEndpoint', 'hello', {}, init);
}

const SpringEndpoint = {
    hello,
};

export default SpringEndpoint;
